In [1]:
import pandas as pd
import seaborn as sns

%matplotlib inline

### Choose the Borough PLUTO set you identify in Problem Set 2 and filter it based on the requirements

In [2]:
df = pd.read_csv("BronxFiltered.csv")

In [3]:
df

Lot  ZipCode                 Address  LotArea  BldgArea  ComArea  \
0         1  10454.0  122 BRUCKNER BOULEVARD    15000         0        0   
1         4  10454.0  126 BRUCKNER BOULEVARD    13770       752      752   
2        10  10454.0  138 BRUCKNER BOULEVARD    35000     39375    39375   
3        17  10454.0  144 BRUCKNER BOULEVARD     2500     12500        0   
4        18  10454.0  148 BRUCKNER BOULEVARD     1875      8595     1719   
...     ...      ...                     ...      ...       ...      ...   
89849   100      NaN                     NaN        0         0        0   
89850   150      NaN                     NaN        0         0        0   
89851   200      NaN                     NaN        0         0        0   
89852  8900      NaN                     NaN        0         0        0   
89853  8900      NaN                     NaN        0         0        0   

       ResArea  OfficeArea  RetailArea  GarageArea  ...  BldgFront  BldgDepth  \
0            0           0           0           0  ...        0.0        0.0   
1            0         272           0           0  ...       16.0       16.0   
2            0           0           0           0  ...      175.0      200.0   
3        12500           0           0           0  ...       25.0       85.0   
4         6876           0        1719           0  ...       25.0       70.0   
...        ...         ...         ...         ...  ...        ...        ...   
89849        0           0           0           0  ...        0.0        0.0   
89850        0           0           0           0  ...        0.0        0.0   
89851        0           0           0           0  ...        0.0        0.0   
89852        0           0           0           0  ...        0.0        0.0   
89853        0           0           0           0  ...        0.0        0.0   

       YearBuilt  YearAlter1  BuiltFAR  ResidFAR  CommFAR  FacilFAR  \
0              0           0      0.00      6.02      5.0       6.5   
1           1931        1994      0.05      6.02      5.0       6.5   
2           1931           0      1.13      6.02      5.0       6.5   
3           1931        2001      5.00      6.02      5.0       6.5   
4           1920        2009      4.58      6.02      5.0       6.5   
...          ...         ...       ...       ...      ...       ...   
89849          0           0      0.00      0.00      0.0       0.0   
89850          0           0      0.00      6.02      3.4       6.5   
89851          0           0      0.00      0.00      0.0       0.0   
89852          0           0      0.00      0.00      0.0       0.0   
89853          0           0      0.00      0.00      0.0       0.0   

          XCoord    YCoord  
0      1005957.0  232162.0  
1      1006076.0  232156.0  
2      1006187.0  232036.0  
3      1006299.0  232033.0  
4      1006363.0  232040.0  
...          ...       ...  
89849        NaN       NaN  
89850        NaN       NaN  
89851        NaN       NaN  
89852        NaN       NaN  
89853        NaN       NaN  

[89854 rows x 27 columns]

### Cardinality check

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89854 entries, 0 to 89853
Data columns (total 27 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Lot         89854 non-null  int64  
 1   ZipCode     89525 non-null  float64
 2   Address     89785 non-null  object 
 3   LotArea     89854 non-null  int64  
 4   BldgArea    89854 non-null  int64  
 5   ComArea     89854 non-null  int64  
 6   ResArea     89854 non-null  int64  
 7   OfficeArea  89854 non-null  int64  
 8   RetailArea  89854 non-null  int64  
 9   GarageArea  89854 non-null  int64  
 10  StrgeArea   89854 non-null  int64  
 11  FactryArea  89854 non-null  int64  
 12  OtherArea   89854 non-null  int64  
 13  NumBldgs    89854 non-null  int64  
 14  NumFloors   89854 non-null  float64
 15  LotFront    89854 non-null  float64
 16  LotDepth    89854 non-null  float64
 17  BldgFront   89854 non-null  float64
 18  BldgDepth   89854 non-null  float64
 19  YearBuilt   89854 non-nul

### Checking Address feature for cardinality since it's categorical

In [5]:
df["Address"].describe()

count           89785
unique          87017
top       SHORE DRIVE
freq               42
Name: Address, dtype: object

### Observation: Total count is 89785 but unique count (cardinality) is 87017